include/nbl/asset/IMeshPacker.h

```diff
@@ -305,29 +258,119 @@ class IMeshPacker : public MeshPackerBase
         core::vector<Triangle> triangles;
     };
 
-    virtual core::vector<TriangleBatch> constructTriangleBatches(MeshBufferType& meshBuffer) = 0;
+    core::vector<TriangleBatch> constructTriangleBatches(MeshBufferType& meshBuffer)
+    {
+        const size_t idxCnt = meshBuffer.getIndexCount();
+        const uint32_t triCnt = idxCnt / 3;
+        _NBL_DEBUG_BREAK_IF(idxCnt % 3 != 0);
 
-protected:
-    //output mesh buffers data
-    SVertexInputParams m_outVtxInputParams;
+        const uint32_t batchCount = (triCnt + m_maxTriangleCountPerMDIData - 1) / m_maxTriangleCountPerMDIData;
```

Comment:
I think this upper bound ought to be `(triCnt-1u)/MIN+1u` (not rounded up based on "always largest batch" but "always smallest batch")

Reply:
you can then resize/trim to actual size

Reply:
TODO

Reply:
Fucking hell why did I have to correct it!?